src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -209,6 +218,21 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
                     isSolutionAvailable,
                     deferredProjectsData.PackageSpecs,
                     token);
+
+                // TODO: To limit risk, we only publish the event when there is a cross-platform PackageReference
+                // project in the solution. Extending this behavior to all solutions is tracked here:
+                // https://github.com/NuGet/Home/issues/4478
+#if !VS14
+                if (projects.OfType<CpsPackageReferenceProject>().Any() &&
+                    !string.IsNullOrEmpty(_dependencyGraphProjectCacheHash))
```

Comment:
Even on the first restore, the hash is set before this during the invocation of `RestorePackageSpecProjectsAsync`:
https://github.com/NuGet/NuGet.Client/blob/7f30828fd6e7aecaa43683ce79068599068ed2bf/src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs#L314